File: Data Structures & Algorithms/validate-parentheses/submission-6.py
class Solution:
    def isValid(self, s: str) -> bool:
        partners = {")": "(", "}": "{", "]": "["}
        stack = []
                   
        for char in s:
            if char not in partners:
                stack.append(char)
            else:
                if stack and stack[-1] == partners[char]:
                    stack.pop()
                else:
                    return False 
            
        return True if not stack else False

File: Data Structures & Algorithms/daily-temperatures/submission-6.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        stack<int> stk;
        vector<int> res(temperatures.size(),0);
        for (int i=0; i<temperatures.size(); i++) {
            while (!stk.empty() && temperatures[i] > temperatures[stk.top()]) {
                int idx = stk.top();
                stk.pop();
                res[idx] = i-idx;
            }
            stk.push(i);
        }
        return res;
    }
};
  